core/drivers/ffa_console.c

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION
+ */
+
+#include <compiler.h>
+#include <console.h>
+#include <drivers/ffa_console.h>
+#include <drivers/serial.h>
+#include <kernel/thread_arch.h>
+
+#define FFA_CONSOLE_LOG_32		(0x8400008A)
+
+static int ffa_console_getchar(struct serial_chip *chip __unused)
+{
+	return -1;
+}
+
+static void ffa_console_putc(struct serial_chip *chip __unused, int ch)
+{
+	thread_hvc(FFA_CONSOLE_LOG_32, 1, ch, 0);
+}
+
+static const struct serial_ops ffa_console_ops = {
+	.getchar = ffa_console_getchar,
```

Comment:
In fact you don't need `.getchar` if it is not implemented so please remove this line and the `ffa_console_getchar()` stub above.

Reply:
Removed `ffa_console_getchar()`